src/xunit.execution/Sdk/Reflection/Reflector.cs

```diff
@@ -33,10 +34,13 @@ public static object[] ConvertArguments(object[] args, Type[] types)
 
             if (args.Length == types.Length)
             {
+                var converted = new List<object>(args.Length);
```

Comment:
What is the purpose of introducing the List<T> here?

Reply:
@bradwilson our comments crossed: https://github.com/xunit/xunit/pull/1546#discussion_r148840099